BASE/src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -674,6 +676,37 @@ public void Flush()
             }
         }
 
+        /// <summary>
+        /// Asynchronously Flushes the in-memory buffer and any metrics being pre-aggregated.
+        /// </summary>
+        /// <returns>The task to await. Task has true set on successful flush.</returns>
+        /// <remarks>
+        /// <a href="https://go.microsoft.com/fwlink/?linkid=525722#flushing-data">Learn more</a>
+        /// </remarks>
+        public Task<bool> FlushAsync(CancellationToken cancellationToken)
+        {
+            if (this.TryGetMetricManager(out MetricManager privateMetricManager))
+            {
+                privateMetricManager.Flush(flushDownstreamPipeline: false);
+            }
+
+            TelemetryConfiguration pipeline = this.configuration;
+            if (pipeline != null)
+            {
+                MetricManager sharedMetricManager = pipeline.GetMetricManager(createIfNotExists: false);
+                sharedMetricManager?.Flush(flushDownstreamPipeline: false);
+
+                ITelemetryChannel channel = pipeline.TelemetryChannel;
+
+                if (channel is IAsyncFlushable)
```

Comment:
Use pattern matching to avoid the extra cast and null check. i.e.
```csharp
if (channel is IAsyncFlushable asyncFlushableChannel)
{
    return asyncFlushableChannel.FlushAsync(cancellationToken);
}
```